### PR TITLE
[LIBNX] USB keyboard support (tested working in Dosbox core and Caprice32 core)

### PR DESCRIPTION
**Pull request by rsn8887**
## Guidelines

1. Rebase before opening a pull request
2. If you are sending several unrelated fixes or features, use a branch and a separate pull request for each
3. If possible try squashing everything in a single commit. This is particularly beneficial in the case of feature merges since it allows easy bisecting when a problem arises

## Description

Add support for USB keyboards to Retroarch Switch

## Related Issues

Allow keyboard entry on physical keyboards connected via USB

## Related Pull Requests

none

## Reviewers

@m4xw

### CODE REVIEW

File: input/input_keymaps.h [hunk 1]
```diff
@@ -65,6 +65,9 @@ extern const struct rarch_key_map rarch_key_map_qnx[];
 extern const struct rarch_key_map rarch_key_map_dos[];
 extern const struct rarch_key_map rarch_key_map_wiiu[];
 extern const struct rarch_key_map rarch_key_map_winraw[];
+#ifdef (HAVE_LIBNX)
```
Comment (m4xw) [inline input/input_keymaps.h]: if defined() or ifdef
Not ifdef ()

File: input/drivers/switch_input.c [hunk 2]
```diff
@@ -64,6 +68,16 @@ static void switch_input_poll(void *data)
          sw->touch_y[i] = touch_position.py;
       }
    }
+
+   for (int i = 0; i < MAX_KEYS; i++) {
+
```
Comment (m4xw) [inline input/drivers/switch_input.c]: Remove spacing

File: input/input_keymaps.h (hunk 1 above)
Comment (rsn8887) [reply input/input_keymaps.h]: fixed

File: input/drivers/switch_input.c (hunk 2 above)
Comment (rsn8887) [reply input/drivers/switch_input.c]: fixed

File: input/drivers/switch_input.c [hunk 3]
```diff
@@ -64,6 +68,20 @@ static void switch_input_poll(void *data)
          sw->touch_y[i] = touch_position.py;
       }
    }
+
+   for (int i = 0; i < SWITCH_NUM_SCANCODES; i++)
```
Comment (bparker06) [inline input/drivers/switch_input.c]: C89 error

Comment (rsn8887) [reply input/drivers/switch_input.c]: What is the error? I am not familiar with C89, but I was able to compile without errors using `make -f Makefile.libnx -j12`

Comment (rsn8887) [reply input/drivers/switch_input.c]: I made a change...
```
int i = 0;
for (i = 0; ...
```
Is that it?

**Comment by rsn8887** [comment]
With my latest changes as of now, I successfully tested this is working using Caprice32 core on Switch.

**Comment by rsn8887** [comment]
Tested working in Dosbox and Caprice32 on Switch.

**Comment by m4xw** [comment]
@rsn8887 In C89 you can't define a variable in a for loop, thats what parker meant.
so it becomes:

```
int i;
for(i = 0;....)
```
It's minor since libnx build explicitly uses gnu11

File: input/drivers/switch_input.c [hunk 4]
```diff
@@ -14,9 +14,13 @@
 
 #define MULTITOUCH_LIMIT 4 /* supports up to this many fingers at once */
 #define TOUCH_AXIS_MAX 0x7fff /* abstraction of pointer coords */
+#define SWITCH_NUM_SCANCODES 114
+
+int switch_keyboard_state[SWITCH_NUM_SCANCODES] = {};
```
Comment (natinusala) [inline input/drivers/switch_input.c]: Move that array to `switch_input_t` for consistency

Comment (rsn8887) [reply input/drivers/switch_input.c]: fixed: changed int array `switch_keyboard_state[SWITCH_NUM_SCANCODES]` to a bool array inside switch_input_t. Renamed to `keyboard_state` for consistency.

**Comment by rsn8887** [comment]
Thanks for the great reviews @bparker06, @m4xw, and @natinusala ! I think it is ready to merge now.

**Comment by rsn8887** [comment]
I have a USB mouse support PR pretty much ready to as soon as this is merged.

**Pull request merged**